mangaki/mangaki/utils/mal.py

```diff
@@ -227,6 +227,30 @@ def _search_works(self, work_type: MALWorks, query: str) -> str:
 
         return xml
 
+    def get_entry_from_work(self, work: Work) -> MALEntry:
+        """
+        Using a mangaki.models.Work to fetch the first (potential) matching MALEntry through MAL Search API.
+
+        WARNING: it is not guaranteed that MAL Search API will return the *good* work
+        (i.e. could be same series, another season, specials, movie, or yet another Japanese invention.)
+
+        Also, will fail on unsupported MALWorks (read the Enum definition to see what is supported).
+
+        :param work: The work to search from (`work.category.slug` and `work.title` will be used)
+        :type work: `mangaki.models.Work`
+        :return: the first matching entry from MAL
+        :rtype: MALEntry
+        """
+        try:
+            mal_work_type = MALWorks(work.category.slug)
+        except ValueError:
+            raise RuntimeError('Unsupported type of work for MAL: {}'.format(work.category))
```

Comment:
`MALWorks` will raise `[x] is not a valid MALWorks`, while this is true, I feel we would do better by raising `Unsupported` rather than `Invalid`, I want to make a separation between Mangaki → MAL errors (Unsupported type of works) and MAL → Mangaki errors (Invalid / new type of works).